File: app/src/main/java/ua/hospes/rtm/domain/race/RaceRepository.java
package ua.hospes.rtm.domain.race;

import android.content.ContentValues;
import android.util.Pair;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;
import ua.hospes.rtm.domain.race.models.RaceItem;

/**
 * @author Andrew Khloponin
 */
public interface RaceRepository {
    Observable<RaceItem> get();

    Observable<List<RaceItem>> listen();

    Observable<RaceItem> listen(int id);

    Observable<Boolean> addNew(RaceItem... items);

    Observable<Boolean> update(List<RaceItem> items);

    Observable<Boolean> updateByTeamId(Iterable<Pair<Integer, ContentValues>> items);

    Single<Integer> remove(RaceItem item);

    Observable<Void> reset();

    Single<Integer> removeAll();
}

File: app/src/main/java/ua/hospes/rtm/domain/race/RaceRepository.kt
package ua.hospes.rtm.domain.race

import kotlinx.coroutines.flow.Flow
import ua.hospes.rtm.domain.race.models.RaceItem

internal interface RaceRepository {
    fun listen(): Flow<List<RaceItem>>
    fun listen(id: Long): Flow<RaceItem>

    suspend fun save(race: RaceItem)
    suspend fun clear()
}